CMakeLists.txt

```diff
@@ -32,32 +32,58 @@ endif()
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_INCLUDE_CURRENT_DIR_IN_INTERFACE ON)
 
-add_definitions(-DHAVE_ZLIB)
+include(FetchContent)
+include(GNUInstallDirs)
+set(INSTALL_CONFIGDIR ${CMAKE_INSTALL_LIBDIR}/cmake/${NIFTI_NIFTILIB_NAME})
+
 
 # add option to build shared libraries. The default is OFF to maintain the
 # current build behavior
+set( 1)
+  set( 1)
+
+option( NIFTI_INSTALL_NO_DEVELOPMENT "ITK build development flag." OFF)
+option( NIFTI_INSTALL_NO_LIBRARIES "ITK build development flag." OFF)
 option( BUILD_SHARED_LIBS "Toggle building shared libraries." OFF)
+if(BUILD_SHARED_LIBS)
+  set(NIFTI_LIBRARY_PROPERTIES
+    BUILD_SHARED_LIBS TRUE
+    POSITION_INDEPENDENT_CODE TRUE)
+  endif()
+
+# Generate a mapping from targets that need to be built to the sub-directories
+# in which they are located.
+macro(INSERT_INTO_DIRMAP _KEY _VALUE)
+  set("modules_${_KEY}" "${_VALUE}")
+endmacro()
+INSERT_INTO_DIRMAP(utils utils)
+INSERT_INTO_DIRMAP(znz znzlib)
+INSERT_INTO_DIRMAP(nifti2io nifti2)
+INSERT_INTO_DIRMAP(niftiio niftilib)
+INSERT_INTO_DIRMAP(cifti cifti)
+INSERT_INTO_DIRMAP(nifticdf nifticdf)
+
+# Fetch and populate the content of each sub-directory to provide access to
+# targets during cmake configure time
+foreach(module nifticdf niftiio nifti2io cifti)
```

Comment:
In order to link to the libraries (in this case libraries in the sub-directories) I thought I required the FetchContent commands. From the cmake FetchContent documentation it seems like this file is a good place to define the behavior of the building of the other libraries. As it stands, I think this can be left out, I just couldn't figure out the pattern to implement that.